### DISCUSSION TITLE
[CI]: not able to create sandbox for given template

### DISCUSSION THREAD

**Discussion by anikethsaha** [codesandbox/codesandbox-client#4220] [General] [answered]
Hi, 

I am trying to integrate codesandbox ci for docsify https://github.com/docsifyjs/docsify/pull/1193

This is `.codesandbox/ci.json` config looks like 

```cjson
{
  "sandboxes": ["docsify-template"],
  "packages": [".", "packages/docsify-server-renderer"]
}
```

The packages are building fine. 👍 
but in `sandboxes` are not being able to create. 

Here is [docsify-template](https://codesandbox.io/s/docsify-template-2d17z) looks like. 

These are working for `vanilla` sandbox

Also, when I am trying to pass a directory [`/docs`](https://github.com/docsifyjs/docsify/tree/develop/docs), it's not working either. 

```cjson
{
  "sandboxes": ["/docs"],
  "packages": [".", "packages/docsify-server-renderer"]
}
```

**Comment by lbogdan** [comment]
Hey @anikethsaha ,

The `sandboxes` field expects an array of sandbox ids, and the sandbox id for your `docsify-template` is `2d17z`, so it should be `"sandboxes": ["2d17z"]`.

**Reply by anikethsaha** [reply]
Thanks @lbogdan for the quick response 🥇 

I changed the config to [this](https://github.com/docsifyjs/docsify/pull/1193/files#diff-3a497e46516b2c9e330c20af04452b77R2)

Still its the [same behavior](https://github.com/docsifyjs/docsify/pull/1193#issuecomment-633024572)

**Comment by lbogdan** [comment, marked as answer]
From what I can see, `package.json` has no `dependecies` field in that sandbox.

The whole idea of CodeSandbox CI is that for every PR build it forks the base sandbox(es) specified in the config and replaces the package(s) in `package.json` -> `dependecies` with the built version(s), so that you can test them.

That being said, I'm not sure how using a base sandbox that doesn't have the package(s) as dependencies can help, TBH.

**Reply by anikethsaha** [reply]
ahh, I see

I need the template so that I dont have to re-create a dummy template. usually, docsify needs an `.html` file and a `.md` to work or to see the behavior. 

From the build, I will be using the published package as a script source `<script src="pkg.csb..." >`
in the template and check the behavior. 


Will it help if I add any dep in `package.json` or it has to be the one which is being tested upon. (`docsify` in my case )?

**Reply by lbogdan** [reply]
Adding an empty `dependencies: {}` should make the forking of the base sandbox work, but I still don't see how that would help, as long as the URL in `<script src="...">` is not replaced with the URL of the built package.

**Reply by anikethsaha** [reply]
> as long as the URL in `<script src="...">` is not replaced with the URL of the built package.

yup, that is the plan, I need to manually replace it every time there is a build to review the change.
